Dockerfile-dev

```diff
@@ -0,0 +1,20 @@
+FROM golang:1.16-buster
+
+RUN apt-get update -y && apt-get install -y build-essential
+
+WORKDIR $GOPATH/src/github.com/weaveworks/weave-gitops
+
+RUN curl -sSfL https://raw.githubusercontent.com/cosmtrek/air/master/install.sh | sh -s
```

Comment:
What is `air`?

Reply:
`air` is a tool that automatically rebuilds a binary when the source code changes. It uses a file named `.air.toml` to specify the configuration, like we can set which folders to watch/exclude.

Reply:
This works pretty well, and doesn't require a new tool and config file:

https://github.com/weaveworks/weave-gitops/blob/993c4723920d968e465958a314fa667ebb952be4/Makefile#L46

Reply:
You may even be able to run `make api-dev` in that container :exploding_head: 

In general though, this approach means there is no visual feedback that a rebuild is in progress.

Reply:
Thanks for the suggestion.

Reply:
Got a bit confused about which tool has no visual feedback. I played around with both tools and seems like air describes more details even when the file cannot be compiled, which find very helpful as we are working in development mode. 

This is the output I got using reflex:
<img width="1045" alt="Screen Shot 2021-09-30 at 12 52 25 AM" src="https://user-images.githubusercontent.com/3892786/135413904-1cae2fef-cfc0-4e41-8d7a-631bfcc9bdbf.png">

This is the output using air:
<img width="683" alt="image" src="https://user-images.githubusercontent.com/3892786/135414087-80d45dcd-52c3-4c1b-af3d-4e9aa555fb5d.png">

This is the change I tested with:
<img width="560" alt="Screen Shot 2021-09-30 at 12 52 18 AM" src="https://user-images.githubusercontent.com/3892786/135414179-d2f218da-b93a-46df-999b-581b452e9b5a.png">

Reply:
I'm happy to change this to use `reflex`.